File: SpringBootWithJpaSpringSecurityOauth/myspringapp/src/main/java/com/springapp/tutotial/controllers/HelloController.java
package com.springapp.tutotial.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

//@Controller
@RestController
public class HelloController {
    @RequestMapping(value="/", method= RequestMethod.GET)
    public String sayHello() {
        return "Hello Hoang, this is the first Rest API response !";
    }
}

File: JavaSpringCourse2022/code/training-app/src/main/kotlin/com/training/controllers/HelloController.kt
package com.training.controllers

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloController {
    @GetMapping("/hello")
    public fun hello(
            @RequestParam(value = "name", defaultValue = "World") name: String?): String? {
        return "Hello $name";
    }
}